### PR TITLE
More detailed hydra test results

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Added some more details to the test results.

image

`KNOWN_FAILING` and `FAILED` results don't show if there are none.

image

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by DizzyEggg** [comment]
What do you think about adding an additional field for tests which failed due to assumption being false? Perhaps ASSUMPTION_FAILED or sth similar?

File: tools/mgba-rom-test-hydra/main.c [hunk 1]
```diff
@@ -9,7 +9,11 @@
  * COMMANDS
  * N: Sets the test name to the remainder of the line.
  * R: Sets the result to the remainder of the line, and flushes any
- *    output buffered since the previous R. */
+ *    output buffered since the previous R.
+ *
+ * //Missing P, F and K documentation, please tell me what to put here lol
+ * 
+ */
```
Comment (mrgriffin) [inline tools/mgba-rom-test-hydra/main.c:16]: Oops, mb!

I suppose something like:
```
P/K/F: Sets the result to the remaining of the line, flushes any output since the previous P/K/F, and for P/K increments the number of passes/known fails.
```

**Comment by AsparagusEduardo** [comment]
> What do you think about adding an additional field for tests which failed due to assumption being false? Perhaps ASSUMPTION_FAILED or sth similar?

Sure thing :)

**Comment by AsparagusEduardo** [comment]
Added `ASSUMPTIONS_FAIL` to the result summary.
Also changed `SKIP` to `ASSUMPTION_FAIL` and made it yellow.

image

File: test/test_runner.c [hunk 2]
```diff
@@ -187,15 +187,22 @@ void CB2_TestRunner(void)
...
             case TEST_RESULT_PASS: result = "PASS"; break;
-            case TEST_RESULT_SKIP: result = "SKIP"; break;
+            case TEST_RESULT_SKIP:
+                result = "ASSUMPTION_FAIL";
+                color = "\e[33m";
+                break;
             case TEST_RESULT_INVALID: result = "INVALID"; break;
             case TEST_RESULT_ERROR: result = "ERROR"; break;
             case TEST_RESULT_TIMEOUT: result = "TIMEOUT"; break;
             default: result = "UNKNOWN"; break;
```
Comment (mrgriffin) [inline test/test_runner.c:197]: At this point, I'd probably just format all the cases on multiple lines :)

**Review by mrgriffin — CHANGES_REQUESTED** [review]

File: test/test_runner.c (hunk 2 above)
Comment (SBird1337) [reply test/test_runner.c:197]: are there other reasons tests might be skipped? For readability I would probably rephrase it to `TEST_RESULT_ASSUMPTION_FAIL` or something along those lines.

Comment (AsparagusEduardo) [reply test/test_runner.c:197]: Yeah, I didn't find other causes for skipping, so it would make sense.

**Comment by AsparagusEduardo** [comment]
Test

**Comment by AsparagusEduardo** [comment]
Ready for re-review

**Review by SBird1337 — APPROVED** [review]

**Review by mrgriffin — APPROVED** [review]
